integrations-examples/ftp/pull_mode/test_pull_from_ftp.py

```
@@ -0,0 +1,372 @@
+            expected_json = os.path.splitext(audio)[0] + ".json"
+            if expected_json not in json_files:
+                unprocessed.append(audio)
+        
+        # Should only include audio2.wav (audio1 and audio3 have JSON files)
+        self.assertEqual(len(unprocessed), 1)
+        self.assertIn("audio2.wav", unprocessed[0])
+    
+    @patch('os.remove')
+    @patch('pull_from_ftp.process_audio_file')
+    @patch('pull_from_ftp.download_file_from_storage')
+    def test_error_handling_in_worker(self, mock_download, mock_process, mock_remove):
```

Ruff (0.12.2)

Unused method argument: `mock_download`

(ARG002)